apps/bfd-server-ng/src/main/java/gov/cms/bfd/server/ng/input/FhirInputConverter.java

```diff
@@ -124,34 +129,50 @@ public static CoverageCompositeId toCoverageCompositeId(@Nullable IdType coverag
   }
 
   /**
-   * Validates an optional _tag {@link TokenParam} if it represents an adjudication status.
+   * Maps a tag code ("Adjudicated" or "PartiallyAdjudicated") to the corresponding ClaimSourceId
+   * enum values.
    *
-   * <p>If the tag is present, this method checks that its value is one of "PartiallyAdjudicated" or
-   * "Adjudicated" (case-insensitive).
-   *
-   * @param tag The _tag {@link TokenParam} from the FHIR request, which can be null.
-   * @return An {@link Optional} containing the validated adjudication status string if the tag was
-   *     present and valid. Returns an empty Optional if the tag was not provided.
-   * @throws InvalidRequestException if the tag is present but its value is blank or unsupported.
+   * @param tag The code from the _tag parameter.
+   * @return A list of matching ClaimSourceId enums.
    */
-  public static String validateTag(@Nullable TokenParam tag) {
+  public static List<ClaimSourceId> getSourceIdsForTagCode(@Nullable TokenParam tag) {
 
-    // A set of all supported adjudication status codes for efficient, case-insensitive lookup.
     Set<String> SUPPORTED_ADJUDICATION_STATUSES =
         Set.of(
             IdrConstants.ADJUDICATION_STATUS_PARTIAL.toUpperCase(),
             IdrConstants.ADJUDICATION_STATUS_FINAL.toUpperCase());
 
-    // Now, validate the non-blank value.
-    if (tag != null && !SUPPORTED_ADJUDICATION_STATUSES.contains(tag.getValue().toUpperCase())) {
+    if (tag == null) {
+      return new ArrayList<>();
+    }
+
+    var systemFromTag = tag.getSystem();
+
+    if (systemFromTag != null && !systemFromTag.equals(SystemUrls.SYS_ADJUDICATION_STATUS)) {
+      throw new InvalidRequestException(
+          "Unsupported system for _tag adjudication status. Expected system '"
+              + SystemUrls.SYS_ADJUDICATION_STATUS
+              + "'.");
+    }
+
+    var statusValue = tag.getValue();
+
+    if (!SUPPORTED_ADJUDICATION_STATUSES.contains(statusValue.toUpperCase())) {
       throw new InvalidRequestException(
           "Unsupported _tag value for adjudication status. Supported values are '"
               + IdrConstants.ADJUDICATION_STATUS_PARTIAL
               + "' and '"
               + IdrConstants.ADJUDICATION_STATUS_FINAL
               + "'.");
     }
+    var matchingSources =
+        Stream.of(ClaimSourceId.values())
+            .filter(
+                sourceId ->
+                    sourceId.getAdjudicationStatus().isPresent()
+                        && sourceId.getAdjudicationStatus().get().equalsIgnoreCase(statusValue))
+            .toList();
 
-    return "";
+    return matchingSources;
```

Comment:
I'd directly return the list instead of setting matchingSources. I think you can also improve this with something like:

```
return Arrays.stream(ClaimSourceId.values())
            .filter(sourceId -> sourceId.getAdjudicationStatus()
                    .map(status -> status.equalsIgnoreCase(statusValue))
                    .orElse(false))
            .toList();
```